src/um/portabledeviceapi.rs

```diff
@@ -0,0 +1,282 @@
+}}
+RIDL!{#[uuid(0xfd8878ac, 0xd841, 0x4d17, 0x89, 0x1c, 0xe6, 0x82, 0x9c, 0xdb, 0x69, 0x34)]
+interface IPortableDeviceResources(IPortableDeviceResourcesVtbl): IUnknown(IUnknownVtbl) {
+    fn GetSupportedResources(
+        pszObjectID: LPCWSTR,
+        ppKeys: *mut *mut IPortableDeviceKeyCollection,
+    ) -> HRESULT,
+    fn GetResourceAttributes(
+        pszObjectID: LPCWSTR,
+        Key: REFPROPERTYKEY,
+        ppResourceAttributes: *mut *mut IPortableDeviceValues,
+    ) -> HRESULT,
+    fn GetStream(
+        pszObjectID: LPCWSTR,
+        Key: REFPROPERTYKEY,
+        dwMode: DWORD,
+        pdwOptimalBufferSize: *mut DWORD,
+        ppStream: *mut *mut IStream,
+    ) -> HRESULT,
+    fn Delete(
+        pszObjectID: LPCWSTR,
+        pKeys: *mut IPortableDeviceKeyCollection,
+    ) -> HRESULT,
+    fn Cancel() -> HRESULT,
+    fn CreateResource(
+        pResourceAttributes: *mut IPortableDeviceValues,
+        ppData: *mut *mut IStream,
+        pdwOptimalWriteBufferSize: *mut DWORD,
+        ppszCookie: *mut LPWSTR,
+    ) -> HRESULT,
+}}
+RIDL!{#[uuid(0x2c8c6dbf, 0xe3dc, 0x4061, 0xbe, 0xcc, 0x85, 0x42, 0xe8, 0x10, 0xd1, 0x26)]
+interface IPortableDeviceCapabilities(IPortableDeviceCapabilitiesVtbl): IUnknown(IUnknownVtbl) {
+    fn GetSupportedCommands(
+        ppCommands: *mut *mut IPortableDeviceKeyCollection,
+    ) -> HRESULT,
+    fn GetCommandOptions(
+        Command: REFPROPERTYKEY,
+        ppOptions: *mut *mut IPortableDeviceValues,
+    ) -> HRESULT,
+    fn GetFunctionalCategories(
+        ppCategories: *mut *mut IPortableDevicePropVariantCollection,
+    ) -> HRESULT,
+    fn GetFunctionalObjects(
+        Category: REFGUID,
+        ppObjectIDs: *mut *mut IPortableDevicePropVariantCollection,
+    ) -> HRESULT,
+    fn GetSupportedContentTypes(
+        Category: REFGUID,
+        ppContentTypes: *mut *mut IPortableDevicePropVariantCollection,
+    ) -> HRESULT,
+    fn GetSupportedFormats(
+        ContentType: REFGUID,
+        ppFormats: *mut *mut IPortableDevicePropVariantCollection,
+    ) -> HRESULT,
+    fn GetSupportedFormatProperties(
+        Format: REFGUID,
+        ppKeys: *mut *mut IPortableDeviceKeyCollection,
+    ) -> HRESULT,
+    fn GetFixedPropertyAttributes(
+        Format: REFGUID,
+        Key: REFPROPERTYKEY,
+        ppAttributes: *mut *mut IPortableDeviceValues,
+    ) -> HRESULT,
+    fn Cancel() -> HRESULT,
+    fn GetSupportedEvents(
+        ppEvents: *mut *mut IPortableDevicePropVariantCollection,
+    ) -> HRESULT,
+    fn GetEventOptions(
+        Event: REFGUID,
+        ppOptions: *mut *mut IPortableDeviceValues,
+    ) -> HRESULT,
+}}
+RIDL!{#[uuid(0xa8792a31, 0xf385, 0x493c, 0xa8, 0x93, 0x40, 0xf6, 0x4e, 0xb4, 0x5f, 0x6e)]
+interface IPortableDeviceEventCallback(IPortableDeviceEventCallbackVtbl): IUnknown(IUnknownVtbl) {
+    fn OnEvent(
+        pEventParameters: *mut IPortableDeviceValues,
+    ) -> HRESULT,
+}}
+extern {
+    pub static CLSID_PortableDevice: CLSID;
```

Comment:
There should never be extern statics, as their values are known and always constant and therefore can be defined as normal constants.